allensdk/brain_observatory/ecephys/stimulus_table/naming_utilities.py

```diff
@@ -195,4 +195,24 @@ def map_column_names(table, name_map=None, ignore_case=True):
     output = table.rename(columns=name_map)
 
     return output
-#
+
+
+def eval_str(val):
+    """Evaluates value if its type==str(numeric) or type==str(list)
+
+    Parameters
+    ----------
+    val: any
+
+    Returns
+    -------
+    val: evaluated val if its type==str(numeric) or type==str(list)
+    or passes the val through if it does not meet the condition
+    """
+
+    if isinstance(val, str):
+        if val.replace('.', '').isdigit():  # checks if val is numeric
+            val = eval(val)
+        elif val[0] == "[" and val[-1] == "]":  # checks if val is list
+            val = tuple(eval(val))
```

Comment:
I separated this to two functions since I think it makes the code better organized through modularization. However, pandas apply is not an efficient way to iterate through all the rows, and running it twice may have additional overhead.